exporter/metrics.go

```diff
@@ -35,26 +119,39 @@ func makeRawMetric(name string, value interface{}) (prometheus.Metric, error) {
 		f = float64(v)
 	case int64:
 		f = float64(v)
+	case float32:
+		f = float64(v)
 	case float64:
 		f = v
-	case primitive.A, primitive.ObjectID: // TODO: Fix this. returning nil, nil just to make tests pass
-		return nil, nil
-
 	case primitive.DateTime:
 		f = float64(v)
-	case primitive.Timestamp:
-		return nil, nil
-
-	case string:
+	case primitive.A, primitive.ObjectID, primitive.Timestamp, string:
 		return nil, nil
-
 	default:
 		return nil, fmt.Errorf("makeRawMetric: unhandled type %T", v)
 	}
 
-	fqName := makeRawMetricName(name)
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
 	help := "TODO"
 	typ := prometheus.UntypedValue
-	d := prometheus.NewDesc(fqName, help, nil, nil)
-	return prometheus.NewConstMetric(d, typ, f)
+
+	fqName, label := rename(prefix, name)
+	if label != "" {
+		labels[label] = name
+	}
+
+	ln := make([]string, 0)
+	lv := make([]string, 0)
+
+	for k, v := range labels {
```

Comment:
looks like we have only one label for now, will we have other labels?

Reply:
This is a recursive function. It might inherit labels.

Reply:
Is it really recursive? I might miss it, but I don't see it now.

Reply:
It will receive more labels in PMM-5757 and in the future it will have labels like cluster information (in plans for the future)